storage/google/cloud/storage/notification.py

```diff
@@ -28,10 +28,12 @@
 NONE_PAYLOAD_FORMAT = 'NONE'
 
 _TOPIC_REF_FMT = '//pubsub.googleapis.com/projects/{}/topics/{}'
-_PROJECT_PATTERN = r'(?P<project>[a-z]+-[a-z]+-\d+)'
+_PROJECT_PATTERN = r'(?P<project>^[a-z][a-z0-9-]{4,28}[a-z0-9]$)'
 _TOPIC_NAME_PATTERN = r'(?P<name>[A-Za-z](\w|[-_.~+%])+)'
 _TOPIC_REF_PATTERN = _TOPIC_REF_FMT.format(
-    _PROJECT_PATTERN, _TOPIC_NAME_PATTERN)
+    _PROJECT_PATTERN.replace(r'^', '').replace(r'$', ''),
```

Comment:
@chemelnucfin These `replace` salls defeat the purpose. I'm confused as to why this is happening.

Reply:
My reasoning was that PROJECT_PATTERN deals with the project only (which is why I chose \A\Z), which should be standalone.  TOPIC_REF_FMT should deal with something else differently, but PROJECT_PATTERN is inside the _TOPIC_REF_FMT.

iirc PROJECT_PATTERN is not used elsewhere now, but it could be in the future.

Reply:
Can you provide a concrete example of what this fixes / what would be broken without it?

Reply:
I changed it back to \A \Z.  
The project pattern would match for test_max_length_fail without any line end delimiters (^$ or \A\Z)
The project pattern would match for test_newline_fail with ^$